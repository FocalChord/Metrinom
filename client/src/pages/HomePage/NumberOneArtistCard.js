import React from "react";
import { CardHeader, Card, CardActionArea, CardMedia, Grid } from "@material-ui/core";
import { useHistory } from "react-router-dom";

const NumberOneArtistCard = ({ artist }) => {
    const history = useHistory();
    return (
        <Grid
            item
            xs={12}
            style={{
                marginTop: 200,
            }}
            container
            direction="column"
            justify="center"
            alignItems="center"
        >
            <Card className={"MuiElevatedCard--01"}>
                <CardHeader
                    className={"MuiCardHeader-root-3"}
                    title={"Your #1 Artist"}
                    classes={{
                        title: "MuiCardHeader-title",
                    }}
                />
                <CardActionArea
                    style={{
                        outline: "none",
                    }}
                    onClick={() => history.push(`/artist/${artist.id}`)}
                >
                    <CardHeader
                        className={"MuiCardHeader-root-1"}
                        title={artist.name}
                        classes={{
                            title: "MuiCardHeader-title",
                        }}
                    />
                    <CardHeader
                        className={"MuiCardHeader-root-2"}
                        title={artist.genres[0]}
                        classes={{
                            title: "MuiCardHeader-title",
                        }}
                    />
                    <CardMedia className={"MuiCardContent-root"} image={artist.images[0].url} />
                </CardActionArea>
            </Card>
        </Grid>
    );
};

export default NumberOneArtistCard;
