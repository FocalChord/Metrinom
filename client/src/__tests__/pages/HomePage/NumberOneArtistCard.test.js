import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import NumberOneArtistCard from "../../../pages/HomePage/NumberOneArtistCard";
import { MemoryRouter } from "react-router-dom";

test("<NumberOneArtistCard />", () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <NumberOneArtistCard artist={{ id: "123", name: "eminem", genres: ["hip hop"], images: [{ url: "img" }] }} />
            </MemoryRouter>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
