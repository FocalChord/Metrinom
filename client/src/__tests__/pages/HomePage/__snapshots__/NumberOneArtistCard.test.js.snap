// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberOneArtistCard /> 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  style={
    Object {
      "marginTop": 200,
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root MuiElevatedCard--01 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root MuiCardHeader-root-3"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Your #1 Artist
        </span>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardHeader-root MuiCardHeader-root-1"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            eminem
          </span>
        </div>
      </div>
      <div
        className="MuiCardHeader-root MuiCardHeader-root-2"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            hip hop
          </span>
        </div>
      </div>
      <div
        className="MuiCardMedia-root MuiCardContent-root"
        style={
          Object {
            "backgroundImage": "url(\\"img\\")",
          }
        }
      />
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
